src/app/digital/rendering/ioobjects/inputs/ConstantNumberRenderer.ts

```diff
@@ -0,0 +1,48 @@
+import {Renderer} from "core/rendering/Renderer";
+import {Line} from "core/rendering/shapes/Line";
+import {Rectangle} from "core/rendering/shapes/Rectangle";
+import {Style} from "core/rendering/Style";
+import {FONT_CONSTANT_NUMBER} from "core/rendering/Styles";
+import {DEFAULT_BORDER_COLOR, DEFAULT_BORDER_WIDTH, DEFAULT_FILL_COLOR, DEFAULT_ON_COLOR, SELECTED_BORDER_COLOR, SELECTED_FILL_COLOR} from "core/utils/Constants";
+import {ConstantNumber} from "digital/models/ioobjects";
+import {V, Vector} from "Vector";
+
+export const ConstantNumberRenderer = (() => {
+
+    // Function to draw the line connecting the 4 outputs
+    const drawOutputConnector = function(renderer: Renderer, size: Vector, borderColor: string): void {
+        const style = new Style(undefined, borderColor, DEFAULT_BORDER_WIDTH);
+        // Y coordinates of the top and bottom
+        const l1 = -(size.y/2)*(1.5);
+        const l2 =  (size.y/2)*(1.5);
+        // X coordinate to draw the vertical line
+        const s = (size.x-DEFAULT_BORDER_WIDTH)/2;
+        // Endpoints of the line to draw
+        const p1 = V(s, l1);
+        const p2 = V(s, l2);
+        renderer.draw(new Line(p1, p2), style);
+    }
+
+    // Function to draw the input value on the component
+    const drawInputText = function(renderer: Renderer, value: number): void {
+        let text = value < 10 ? value.toString() : "ABCDEF".charAt(value - 10);
+        renderer.text(text, V(0,2.5), "center", DEFAULT_ON_COLOR, FONT_CONSTANT_NUMBER);
+    }
+
+    return {
+        render(renderer: Renderer, object: ConstantNumber, selected: boolean): void {
+            const transform = object.getTransform();
+            const fillColor = selected ? SELECTED_FILL_COLOR : DEFAULT_FILL_COLOR;
+            const borderColor = selected ? SELECTED_BORDER_COLOR : DEFAULT_BORDER_COLOR;
+            const style = new Style(fillColor, borderColor, DEFAULT_BORDER_WIDTH);
+            // Draw the rectangle first, subtracting border width for alignment
+            var rectSize = transform.getSize();
+            rectSize.x -= DEFAULT_BORDER_WIDTH;
+            rectSize.y -= DEFAULT_BORDER_WIDTH;
```

Comment:
this can be simplified to `const rectSize = transform.getSize().sub(DEFAULT_BORDER_WIDTH);`